### DISCUSSION TITLE
Resume EC Fan Control

### DISCUSSION THREAD

**Discussion by Hammerfest** [DadouLapous/DellG5SE-Fan-Linux#3] [General]
How can I pass control back to the EC on shutdown?

I have Windows and Linux, when using this (or honestly, possibly another tool as I have tried many, thats why I am not putting it in issues) in Linux its fine, but when I boot back into windows, the Fan's don't seem to kick in when they should, it gets stupid hot despite the "default" fan limits that where in place prior, AND even if I set a custom fan profile in AWCC it only works for a short time before it ignores it unless I toggle it on/off again.

Thoughts?

**Comment by DavidLapous** [comment]
EC variables gets reset on reboot, so this script can't be the culprit (you can send me an ec dump with RWEverything with windows, I'll confirm). I do have the same issue on windows, AWCC fan control did work at some point but it doesn't work now (it broke before I've tried to mess with EC variables...). BTW, the `-r` option, restores without a reboot fan control to the EC.
Reinstalling windows and correct versions of awcc / Dell crap *should* fix fan control. The method my script is using is OS agnostic, so if someone want to rewrite some (`update_vars` and `write_to_ec`) functions of the cpp file, it will work with windows, and you could then remove the dell AWCC crap (I did try updating fans with RWEverything on windows, and it works).
I'm not really using windows,  and as far as I know, there is no `dell-smm-hwmon`  alternative to read temps / fans via the dell sensors on windows, but it could be replaced by reading the correct variables on the EC. That should work fine, but I don't think I'll take the time to do it as I'm not using windows.

**Reply by Hammerfest** [reply]
Well, its good to know im not the only one with the issue and its not my fumbling with linux.
I have the same issue with the fans using the recommended AWCC and latest so I will be opening a ticket with Dell, I recommend anyone else having the issue do so as well, hopefully they will fix it (even if you dont use windows).

Yep, if someone could make this work for windows it would be FANTASTIC, I have 2 of these, one I use windows mainly on as its required for my work, the linux boot is for testing hardware, the other I use windows/linux 50/50.

I did try FanControl as its got a Dell addon but the 2nd sensor is incorrect and it cant control the 2nd fan (I made an issue ticket with that source but hes not interested in fixing right now either)

Sad times, well, at least I got these for cheap ($1100 for the R9, $750 for the R7)
